### ISSUE TITLE
Can I use IMAGE_DATA as input when training GoogleNet?

### ISSUE THREAD

**Issue by shuangao** [BVLC/caffe#1908]
Hi all, I just want to train the [googleNet](https://github.com/BVLC/caffe/tree/master/models/bvlc_googlenet) and using the image data as input data type (giving the path of the training and validation txt file). I failed to train it because I encountered the following problem: the loss becomes 'nan' from the 80th iteration.
````
I0219 15:15:29.717862 14350 solver.cpp:161] Learning Rate Policy: step
I0219 15:15:31.617120 14350 solver.cpp:209] Iteration 0, loss = 11.0746
I0219 15:15:31.617197 14350 solver.cpp:224]     Train net output #0: loss1/loss1 = 6.89612 (* 0.3 = 2.06884 loss)
I0219 15:15:31.617211 14350 solver.cpp:224]     Train net output #1: loss2/loss1 = 6.90393 (* 0.3 = 2.07118 loss)
I0219 15:15:31.617223 14350 solver.cpp:224]     Train net output #2: loss3/loss3 = 6.93463 (* 1 = 6.93463 loss)
I0219 15:15:31.617297 14350 solver.cpp:445] Iteration 0, lr = 0.01
I0219 15:17:06.672966 14350 solver.cpp:209] Iteration 40, loss = 1.54674
I0219 15:17:06.681916 14350 solver.cpp:224]     Train net output #0: loss1/loss1 = 32.7512 (* 0.3 = 9.82536 loss)
I0219 15:17:06.681941 14350 solver.cpp:224]     Train net output #1: loss2/loss1 = 32.7512 (* 0.3 = 9.82536 loss)
I0219 15:17:06.681954 14350 solver.cpp:224]     Train net output #2: loss3/loss3 = 32.7512 (* 1 = 32.7512 loss)
I0219 15:17:06.681965 14350 solver.cpp:445] Iteration 40, lr = 0.01
I0219 15:18:47.157843 14350 solver.cpp:209] Iteration 80, loss = nan
I0219 15:18:47.158116 14350 solver.cpp:224]     Train net output #0: loss1/loss1 = nan (* 0.3 = nan loss)
I0219 15:18:47.158138 14350 solver.cpp:224]     Train net output #1: loss2/loss1 = nan (* 0.3 = nan loss)
I0219 15:18:47.158149 14350 solver.cpp:224]     Train net output #2: loss3/loss3 = nan (* 1 = nan loss)
I0219 15:18:47.158161 14350 solver.cpp:445] Iteration 80, lr = 0.01
I0219 15:20:27.284762 14350 solver.cpp:209] Iteration 120, loss = nan
I0219 15:20:27.285017 14350 solver.cpp:224]     Train net output #0: loss1/loss1 = nan (* 0.3 = nan loss)
I0219 15:20:27.285038 14350 solver.cpp:224]     Train net output #1: loss2/loss1 = nan (* 0.3 = nan loss)
I0219 15:20:27.285051 14350 solver.cpp:224]     Train net output #2: loss3/loss3 = nan (* 1 = nan loss)
````
Here is the data layer
````
layers {
  top: "data"
  top: "label"
  name: "data"
  type: IMAGE_DATA
  image_data_param {
    source: "/scratch/train_new.txt"
    batch_size: 32
    new_height: 256
    new_width: 256
  }
  include {
    phase: TRAIN
  }
  transform_param {
    mirror: true
    crop_size: 224
    mean_file: "./data/ilsvrc12/imagenet_mean.binaryproto"
    # mean_value: 104
    # mean_value: 117
    # mean_value: 123
  }
}
layers {
  top: "data"
  top: "label"
  name: "data"
  type: IMAGE_DATA
  image_data_param {
    source: "/scratch/val_new.txt"
    batch_size: 50
    new_height: 256
    new_width: 256
  }
  include {
    phase: TEST
  }
  transform_param {
    mirror: false
    crop_size: 224
    mean_file: "./data/ilsvrc12/imagenet_mean.binaryproto"
    # mean_value: 104
    # mean_value: 117
    # mean_value: 123
  }
}
````
Then I turned to Leveldb, everything is OK.  
````
I0219 15:30:47.000522 15901 solver.cpp:209] Iteration 0, loss = 11.0459
I0219 15:30:47.000571 15901 solver.cpp:224]     Train net output #0: loss1/loss1 = 6.91178 (* 0.3 = 2.07353 loss)
I0219 15:30:47.000584 15901 solver.cpp:224]     Train net output #1: loss2/loss1 = 6.90464 (* 0.3 = 2.07139 loss)
I0219 15:30:47.000596 15901 solver.cpp:224]     Train net output #2: loss3/loss3 = 6.90094 (* 1 = 6.90094 loss)
I0219 15:30:47.000650 15901 solver.cpp:445] Iteration 0, lr = 0.01
I0219 15:32:00.073784 15901 solver.cpp:209] Iteration 40, loss = 10.2859
I0219 15:32:00.074018 15901 solver.cpp:224]     Train net output #0: loss1/loss1 = 6.20014 (* 0.3 = 1.86004 loss)
I0219 15:32:00.074033 15901 solver.cpp:224]     Train net output #1: loss2/loss1 = 6.45299 (* 0.3 = 1.9359 loss)
I0219 15:32:00.074044 15901 solver.cpp:224]     Train net output #2: loss3/loss3 = 5.91532 (* 1 = 5.91532 loss)
I0219 15:32:00.074056 15901 solver.cpp:445] Iteration 40, lr = 0.01
I0219 15:33:13.134585 15901 solver.cpp:209] Iteration 80, loss = 9.53847
I0219 15:33:13.134856 15901 solver.cpp:224]     Train net output #0: loss1/loss1 = 5.88969 (* 0.3 = 1.76691 loss)
I0219 15:33:13.134889 15901 solver.cpp:224]     Train net output #1: loss2/loss1 = 5.97238 (* 0.3 = 1.79171 loss)
I0219 15:33:13.134910 15901 solver.cpp:224]     Train net output #2: loss3/loss3 = 5.75969 (* 1 = 5.75969 loss)
I0219 15:33:13.134930 15901 solver.cpp:445] Iteration 80, lr = 0.01
I0219 15:34:26.195854 15901 solver.cpp:209] Iteration 120, loss = 9.31518
I0219 15:34:26.196168 15901 solver.cpp:224]     Train net output #0: loss1/loss1 = 5.76563 (* 0.3 = 1.72969 loss)
I0219 15:34:26.196200 15901 solver.cpp:224]     Train net output #1: loss2/loss1 = 5.80312 (* 0.3 = 1.74094 loss)
I0219 15:34:26.196221 15901 solver.cpp:224]     Train net output #2: loss3/loss3 = 5.72176 (* 1 = 5.72176 loss)
````
Did I make any mistakes?

**Comment by rohrbach** [comment]
You could try decreasing the learning rate. A learning rate which is too high can blow up the loss.

However, please ask usage questions on [caffe-users](https://groups.google.com/forum/#!forum/caffe-users) -- the issues tracker is primarily for Caffe development discussion. Thanks!
